web/db/actions/Post.ts

```diff
@@ -83,23 +90,46 @@ async function getResizedUploadUrl(uuid: string): Promise<string> {
 }
 
 async function deleteAttachments(keysToDelete: string[]) {
-  //will make key splitting more efficient later
-  const objectsToDelete = keysToDelete.map((keyToDelete) => ({
-    Key:
-      keyToDelete.split("/")[keyToDelete.split("/").length - 2] +
-      "/" +
-      keyToDelete.split("/")[keyToDelete.split("/").length - 1],
-  }));
-  const deleteObjectsCommand = new DeleteObjectsCommand({
-    Bucket: consts.storageBucket,
-    Delete: { Objects: objectsToDelete },
-  });
-  try {
-    const returned = await storageClient.send(deleteObjectsCommand);
-    return { success: true };
-  } catch (error) {
-    return { success: false };
+  for (let i = 0; i < 2; i++) {
+    const objectsToDelete = keysToDelete.map((keyToDelete) => ({
+      Key: keyToDelete,
+    }));
+    const deleteObjectsCommand = new DeleteObjectsCommand({
+      Bucket: consts.storageBucket,
+      Delete: { Objects: objectsToDelete, Quiet: true },
+    });
+    try {
+      const returned = await storageClient.send(deleteObjectsCommand);
+      let arr: string[] = [];
+      if (returned["Deleted"]) {
+        returned["Deleted"].forEach((element) => {
+          element["Key"] &&
+            !element["DeleteMarker"] &&
+            arr.push(element["Key"]);
+        });
+        keysToDelete = arr;
+      } else {
+        return { success: true };
+      }
+    } catch (error) {
+      return { success: false };
```

Comment:
throw an error on failure

Reply:
You can also consider adding a `TODO` comment mentioning that we may benefit from writing a cron job to mark unsuccessful deletions to delete later